### PR TITLE
[Snyk] Sanitize and bind "common/common-Func" queries

### PR DESCRIPTION
**Pull request by ataghzout**
## Description
Sanitize and bind "common/common-Func" related queries to avoid sql injections in the future.

**Fixes** # MON-12873

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [x] 21.04.x
- [x] 21.10.x
- [x] 22.04.x
- [] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/common/common-Func.php [hunk 1]
```diff
@@ -673,17 +676,22 @@ function getMyHostExtendedInfoImage($host_id, $field, $flag1stLevel = null, $ant
...
     if (isset($flag1stLevel) && $flag1stLevel) {
-        $rq = "SELECT ehi.`" . $field . "` " .
+        $rq = "SELECT :field " .
             "FROM extended_host_information ehi " .
-            "WHERE ehi.host_host_id = '" . CentreonDB::escape($host_id) . "' LIMIT 1";
-        $DBRESULT = $pearDB->query($rq);
-        $row = $DBRESULT->fetchRow();
+            "WHERE ehi.host_host_id = :host_host_id LIMIT 1";
+        $statement = $pearDB->prepare($rq);
+        $statement->bindValue(':field', $field, \PDO::PARAM_STR);
```
Comment (callapa) [inline www/include/common/common-Func.php:683]: Use bindColumn instead

**Review by callapa — CHANGES_REQUESTED** [review]

File: www/include/common/common-Func.php [hunk 2]
```diff
@@ -2386,14 +1573,6 @@ function getDuplicateNumberOption()
...
 }
 
-function isNotEmptyAfterStringSanitize($test): bool
-{
-    if (empty(filter_var($test, FILTER_SANITIZE_STRING))) {
-        return false;
-    } else {
-        return true;
-    }
-}
```
Comment (jeremyjaouen) [inline www/include/common/common-Func.php]: Please restore this method as it is still used 

```
2022-06-23T17:56:54.350068273Z [23-Jun-2022 19:56:54 Europe/Paris] PHP Fatal error:  Uncaught Error: Call to undefined function isNotEmptyAfterStringSanitize() in /usr/share/centreon/vendor/openpsa/quickform/lib/HTML/QuickForm/Rule/Callback.php:57
```

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by kduret — COMMENTED** [review]
@ataghzout you need to resolve conflicts

File: www/include/common/common-Func.php [hunk 3]
```diff
@@ -2224,35 +1440,6 @@ function array_column($array, $column_name)
 
 }
 
-/**
- * Get current Centreon version
- */
-function getCentreonVersion($pearDB)
```
Comment (callapa) [inline www/include/common/common-Func.php:2230]: Do not delete this function, it is still in use

**Review by callapa — CHANGES_REQUESTED** [review]

File: www/include/common/common-Func.php [hunk 4]
```diff
@@ -2224,35 +1504,6 @@ function array_column($array, $column_name)
...
```
Comment (jeremyjaouen) [inline www/include/common/common-Func.php:2230]: this method should not be deleted

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

File: www/include/common/common-Func.php [hunk 5]
```diff
@@ -524,15 +285,22 @@ function getMyHostExtendedInfoFieldFromMultiTemplates($host_id, $field)
...
-        $rq2 = "SELECT ehi.`" . $field . "` " .
-            "FROM extended_host_information ehi " .
-            "WHERE ehi.host_host_id = '" . $row['host_tpl_id'] . "' LIMIT 1";
-        $DBRESULT2 = $pearDB->query($rq2);
-        $row2 = $DBRESULT2->fetchRow();
+    $statement = $pearDB->prepare($rq);
+    $statement->bindValue(':host_host_id', (int)$host_id, \PDO::PARAM_INT);
+    $statement->execute();
+    while ($row = $statement->fetchRow()) {
+        $rq2 = "SELECT ehi." . ":field" .
```
Comment (kduret) [inline www/include/common/common-Func.php]: I think it's not possible to do this

File: www/include/common/common-Func.php [hunk 6]
```diff
@@ -448,12 +252,16 @@ function getMyHostField($host_id, $field)
...
-    $DBRESULT = $pearDB->query($query);
-    while ($row = $DBRESULT->fetchRow()) {
-        $DBRESULT2 = $pearDB->query("SELECT `" . $field . "` FROM host WHERE host_id = '" . $row['host_tpl_id'] . "'");
-        while ($row2 = $DBRESULT2->fetchRow()) {
+    $query = "SELECT host_tpl_id FROM host_template_relation WHERE host_host_id = :host_id ORDER BY `order` ASC";
+    $statement = $pearDB->prepare($query);
+    $statement->bindValue(':host_id', (int) $host_id, \PDO::PARAM_INT);
+    $statement->execute();
+    while ($row = $statement->fetchRow()) {
+        $statement2 = $pearDB->prepare("SELECT :field  FROM host WHERE host_id = :host_tpl_id");
```
Comment (kduret) [inline www/include/common/common-Func.php]: I think it's not possible to bind `:field`

File: www/include/common/common-Func.php [hunk 7]
```diff
@@ -681,11 +430,14 @@ function getMyHostExtendedInfoField($host_id, $field)
     }
     global $pearDB, $oreon;
 
-    $rq = "SELECT ehi.`" . $field . "` " .
+    $rq = "SELECT ehi." . ":field" . " " .
```
Comment (kduret) [inline www/include/common/common-Func.php]: I think it's not possible to do this

File: www/include/common/common-Func.php [hunk 8]
```diff
@@ -702,17 +454,22 @@ function getMyHostExtendedInfoImage($host_id, $field, $flag1stLevel = null, $ant
     }
 
     if (isset($flag1stLevel) && $flag1stLevel) {
-        $rq = "SELECT ehi.`" . $field . "` " .
+        $rq = "SELECT :field " .
```
Comment (kduret) [inline www/include/common/common-Func.php]: I think it's not possible to do this

File: www/include/common/common-Func.php [hunk 9]
```diff
@@ -725,21 +482,28 @@ function getMyHostExtendedInfoImage($host_id, $field, $flag1stLevel = null, $ant
...
-        while ($row = $DBRESULT->fetchRow()) {
-            $rq2 = "SELECT ehi.`" . $field . "` " .
-                "FROM extended_host_information ehi " .
-                "WHERE ehi.host_host_id = '" . $row['host_tpl_id'] . "' LIMIT 1";
-            $DBRESULT2 = $pearDB->query($rq2);
-            $row2 = $DBRESULT2->fetchRow();
+        $htStatement = $pearDB->prepare($rq);
+        $htStatement->bindValue(':host_host_id', (int) $host_id, \PDO::PARAM_INT);
+        $htStatement->execute();
+        $rq2 = "SELECT :field " .
```
Comment (kduret) [inline www/include/common/common-Func.php]: I think it's not possible to do this

**Review by kduret — CHANGES_REQUESTED** [review]

**Comment by Thebarda** [comment]
293 files changes. I hope this is a failed merge

**Comment by emabassi-ext** [comment]
> 293 files changes. I hope this is a failed merge

Fixed !

File: www/include/common/common-Func.php [hunk 10]
```diff
@@ -80,16 +80,6 @@ function returnSvg(string $svgPath, string $color, float $height, float $width):
     }
 }
 
-/*
- * Form Rules
- */
-function slash($elem = null)
```
Comment (callapa) [inline www/include/common/common-Func.php]: This function is used in this file `www/include/Administration/parameters/general/form.php`

**Review by callapa — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

File: www/include/common/common-Func.php [hunk 11]
```diff
@@ -578,53 +353,30 @@ function getMyHostMacroFromMultiTemplates($host_id, $field)
...
-        $macroValue = str_replace("#BS#", "\\", $macroValue);
-        return $macroValue;
-    } else {
-        return getMyHostMacroFromMultiTemplates($host_id, $field);
-    }
-}
-
 function getMyServiceCategories($service_id = null)
 {
     global $pearDB, $oreon;
```
Comment (TamazC) [inline www/include/common/common-Func.php:358]:
```suggestion
    global $pearDB;
```

Comment (TamazC) [reply www/include/common/common-Func.php:358]: $oreon is never used

File: www/include/common/common-Func.php [hunk 12]
```diff
@@ -657,13 +409,15 @@ function getMyServiceMacro($service_id, $field)
         return;
     }
     global $pearDB, $oreon;
```
Comment (TamazC) [inline www/include/common/common-Func.php]:
```suggestion
    global $pearDB,;
```
$oreon is never used

File: www/include/common/common-Func.php [hunk 13]
```diff
@@ -1060,10 +735,11 @@ function getMyServiceField($service_id, $field)
...
-        $query = "SELECT `" . $field . "`, service_template_model_stm_id FROM service WHERE service_id = '" .
-            CentreonDB::escape($service_id) . "' LIMIT 1";
-        $DBRESULT = $pearDB->query($query);
-        $row = $DBRESULT->fetchRow();
+        $statement = $pearDB->prepare("SELECT `" . $field . "` , service_template_model_stm_id " .
+            "FROM service WHERE service_id = :service_id LIMIT 1") ;
+        $statement->bindValue(':service_id', (int) $service_id, \PDO::PARAM_INT);
+        $statement->execute();
+        $row = $statement->fetchRow();
         $field_result = $row[$field];
```
Comment (TamazC) [inline www/include/common/common-Func.php]:
```suggestion

```
unused variable $field_result

File: www/include/common/common-Func.php [hunk 14]
```diff
@@ -1112,43 +791,6 @@ function getMyServiceExtendedInfoField($service_id, $field)
     }
 }
 
-function getMyServiceExtendedInfoImage($service_id, $field)
```
Comment (TamazC) [inline www/include/common/common-Func.php:1115]: This funcion is still used in :
- www\include\configuration\configObject\service\listServiceByHost.php:346
- www\include\configuration\configObject\service\listServiceByHostGroup.php:313
- www\include\configuration\configObject\service_template_model\listServiceTemplateModel.php:219
- www\include\monitoring\objectDetails\serviceDetails.php:872
- www\include\monitoring\objectDetails\serviceDetails.php:877

File: www/include/common/common-Func.php [hunk 15]
```diff
@@ -1672,147 +1207,23 @@ function getMyHostID($host_name = null)
...
-    }
-    $DBRESULT = $pearDB->query("SELECT graph_id FROM giv_graphs_template LIMIT 1");
-    if ($DBRESULT->rowCount()) {
-        $gt = $DBRESULT->fetchRow();
-        return $gt["graph_id"];
-    }
-    return null;
-}
-
 function getDefaultGraph($service_id = null, $rrdType = null)
```
Comment (TamazC) [inline www/include/common/common-Func.php]: The function "getDefaultGraph" is unused, so you could remove it.

File: www/include/common/common-Func.php [hunk 16]
```diff
@@ -1911,127 +1324,36 @@ function service_has_graph($host, $service, $dbo = null)
...
-    }
-    return null;
-}
-
 /*
  * function getNDOPrefix()
  * - This function return NDOPrefix tables.
  *
  * @return	string	$conf_ndo["db_prefix"]	(string contains prefix like "nagios_")
  */
```
Comment (TamazC) [inline www/include/common/common-Func.php]:
```suggestion
 */

```

**Review by TamazC — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

File: www/include/common/common-Func.php [hunk 17]
```diff
@@ -1573,15 +1134,18 @@ function getMyCheckCmdName($service_id = null)
...
 
+    $query = "SELECT command_name FROM command WHERE command_id = :command_id LIMIT 1";
+    $statement = $pearDB->prepare($query);
     while (1) {
-        $query = "SELECT command_command_id, service_template_model_stm_id FROM service " .
-            "WHERE service_id = '" . CentreonDB::escape($service_id) . "' LIMIT 1";
-        $DBRESULT = $pearDB->query($query);
-        $row = $DBRESULT->fetchRow();
+        $statement2 = $pearDB->prepare("SELECT command_command_id, service_template_model_stm_id FROM service " .
+            "WHERE service_id = :service_id LIMIT 1") ;
```
Comment (kduret) [inline www/include/common/common-Func.php]:
```suggestion
            "WHERE service_id = :service_id LIMIT 1");
```

File: www/include/common/common-Func.php [hunk 18]
```diff
@@ -1060,11 +735,11 @@ function getMyServiceField($service_id, $field)
     $tab = array();
 
     while (1) {
-        $query = "SELECT `" . $field . "`, service_template_model_stm_id FROM service WHERE service_id = '" .
-            CentreonDB::escape($service_id) . "' LIMIT 1";
-        $DBRESULT = $pearDB->query($query);
-        $row = $DBRESULT->fetchRow();
-        $field_result = $row[$field];
+        $statement = $pearDB->prepare("SELECT `" . $field . "` , service_template_model_stm_id " .
+            "FROM service WHERE service_id = :service_id LIMIT 1") ;
```
Comment (kduret) [inline www/include/common/common-Func.php]:
```suggestion
            "FROM service WHERE service_id = :service_id LIMIT 1");
```

File: www/include/common/common-Func.php [hunk 19]
```diff
@@ -1158,10 +799,11 @@ function getMyServiceName($service_id = null)
     $tab = array();
 
     while (1) {
-        $query = "SELECT service_description, service_template_model_stm_id FROM service WHERE service_id = '" .
-            CentreonDB::escape($service_id) . "' LIMIT 1";
-        $DBRESULT = $pearDB->query($query);
-        $row = $DBRESULT->fetchRow();
+        $statement = $pearDB->prepare("SELECT service_description, service_template_model_stm_id FROM service " .
+            " WHERE service_id = :service_id LIMIT 1") ;
```
Comment (kduret) [inline www/include/common/common-Func.php]:
```suggestion
            " WHERE service_id = :service_id LIMIT 1");
```

File: www/include/common/common-Func.php [hunk 20]
```diff
@@ -1186,10 +828,11 @@ function getMyServiceAlias($service_id = null)
     $tab = array();
 
     while (1) {
-        $query = "SELECT service_alias, service_template_model_stm_id FROM service " .
-            "WHERE service_id = '" . CentreonDB::escape($service_id) . "' LIMIT 1";
-        $DBRESULT = $pearDB->query($query);
-        $row = $DBRESULT->fetchRow();
+        $statement = $pearDB->prepare("SELECT service_alias, service_template_model_stm_id FROM service " .
+            "WHERE service_id = :service_id LIMIT 1") ;
```
Comment (kduret) [inline www/include/common/common-Func.php]:
```suggestion
            "WHERE service_id = :service_id LIMIT 1");
```

File: www/include/common/common-Func.php [hunk 21]
```diff
@@ -1213,11 +856,12 @@ function getMyServiceGraphID($service_id = null)
...
     $tab = array();
     while (1) {
-        $query = "SELECT esi.graph_id, service_template_model_stm_id FROM service, extended_service_information esi " .
-            "WHERE service_id = '" . CentreonDB::escape($service_id) .
-            "' AND esi.service_service_id = service_id LIMIT 1";
-        $DBRESULT = $pearDB->query($query);
-        $row = $DBRESULT->fetchRow();
+        $statement = $pearDB->prepare("SELECT esi.graph_id, service_template_model_stm_id" .
+            " FROM service, extended_service_information esi " .
+            "WHERE service_id = :service_id AND esi.service_service_id = service_id LIMIT 1") ;
```
Comment (kduret) [inline www/include/common/common-Func.php]:
```suggestion
            "WHERE service_id = :service_id AND esi.service_service_id = service_id LIMIT 1");
```

File: www/include/common/common-Func.php [hunk 22]
```diff
@@ -1236,12 +880,15 @@ function getMyServiceGraphID($service_id = null)
...
-    $query = "SELECT s.service_id FROM services s " .
-        " WHERE (s.description = '" . $dbb->escape($service_description) . "'
-                        OR s.description = '" . $dbb->escape(utf8_encode($service_description)) . "') "
-        . " AND s.host_id = " . $dbb->escape($host_id) . " LIMIT 1";
-    $DBRESULT = $dbb->query($query);
-    $row = $DBRESULT->fetchRow();
+    $statement = $dbb->prepare("SELECT s.service_id FROM services s " .
+        " WHERE (s.description = :service_description
+                        OR s.description = :utf8_uncoded_service_description ) "
+        . " AND s.host_id = :host_id LIMIT 1") ;
```
Comment (kduret) [inline www/include/common/common-Func.php]:
```suggestion
        . " AND s.host_id = :host_id LIMIT 1");
```

**Review by kduret — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**